stl/inc/span

```diff
@@ -0,0 +1,729 @@
+ && _RANGES sized_range<_Rng>
+ && (_RANGES safe_range<_Rng> || is_const_v<_Ty>)
+ && is_convertible_v<remove_reference_t<_RANGES range_reference_t<_Rng>>(*)[], _Ty(*)[]>;
+// clang-format on
+#else // ^^^ __cpp_lib_concepts / !__cpp_lib_concepts vvv
+
+// STRUCT TEMPLATE _Is_span
+template <class>
+struct _Is_span : false_type {};
+
+template <class _Ty, size_t _Extent>
+struct _Is_span<span<_Ty, _Extent>> : true_type {};
+
+// STRUCT TEMPLATE _Is_std_array
+template <class>
+struct _Is_std_array : false_type {};
+
+template <class _Ty, size_t _Size>
+struct _Is_std_array<array<_Ty, _Size>> : true_type {};
+
+// STRUCT TEMPLATE _Is_span_convertible_range
+template <class _Rng, class _Ty>
+struct _Is_span_convertible_range
+    : bool_constant<is_convertible_v<remove_pointer_t<decltype(_STD data(_STD declval<_Rng&>()))> (*)[], _Ty (*)[]>> {};
+
+// STRUCT TEMPLATE _Has_container_interface
+template <class, class = void>
+struct _Has_container_interface : false_type {};
+
+template <class _Rng>
+struct _Has_container_interface<_Rng,
+    void_t<decltype(_STD data(_STD declval<_Rng&>())), decltype(_STD size(_STD declval<_Rng&>()))>> : true_type {};
+
+// STRUCT TEMPLATE _Is_span_compatible_range
+// clang-format off
+template <class _Rng, class _Ty>
+struct _Is_span_compatible_range : bool_constant<conjunction_v<
+    negation<is_array<_Rng>>,
+    negation<_Is_span<remove_const_t<_Rng>>>,
+    negation<_Is_std_array<remove_const_t<_Rng>>>,
+    _Has_container_interface<_Rng>,
+    _Is_span_convertible_range<_Rng, _Ty>>> {};
+// clang-format on
+
+#endif // !__cpp_lib_concepts
+
+// [views.span]
+// CLASS TEMPLATE span
+template <class _Ty, size_t _Extent = dynamic_extent>
+class span : public _Span_extent_type<_Extent> {
+public:
+    using element_type           = _Ty;
+    using value_type             = remove_cv_t<_Ty>;
+    using size_type              = size_t;
+    using difference_type        = ptrdiff_t;
+    using pointer                = _Ty*;
+    using const_pointer          = const _Ty*;
+    using reference              = _Ty&;
+    using const_reference        = const _Ty&;
+    using iterator               = _Span_iterator<_Ty>;
+    using const_iterator         = _Span_iterator<const _Ty>;
+    using reverse_iterator       = _STD reverse_iterator<iterator>;
+    using const_reverse_iterator = _STD reverse_iterator<const_iterator>;
+
+    static constexpr size_type extent = _Extent;
+
+    // [span.cons] Constructors, copy, and assignment
+#ifdef __cpp_lib_concepts
+
+    // clang-format off
+    constexpr span() noexcept requires (_Extent == 0 || _Extent == dynamic_extent) = default;
+
+    template <_Is_span_compatible_iterator<element_type> _It>
+    constexpr span(_It _First, size_type _Count) noexcept // strengthened
+        : _Mybase(_Count), _Mydata(_STD to_address(_First)) {
+#if _CONTAINER_DEBUG_LEVEL > 0
+        (void) (_First + _Count); // Implicit bounds checking for checked iterators

```

Comment:
This:

* Shouldn't be guarded by `_CONTAINER_DEBUG_LEVEL` so it can properly support checked-in-release iterators,

* Shouldn't be unconditional, either, since it's user-observable

I'm inclined to suggest putting `_Mydata(_STD to_address(_Get_unwrapped_n(_First, _Count)))` in the mem-initializer. @BillyONeal, does that sound right? (Should we file an issue to put conditional `noexcept` on all of the unwrapping/checking machinery for better debug codegen?)

Reply:
That sounds right :)

Reply:
went with the `_Mydata(_STD to_address(_Get_unwrapped_n(_First, _Count)))`